torch/_inductor/codegen/simd.py

```diff
@@ -1544,7 +1603,21 @@ def _codegen_mix_order_reduction(self, node1, node2):
         ):
             return self._codegen_mix_order_reduction(node2, node1)
 
-        # pyrefly: ignore [bad-assignment]
+        def _pick_split_size():
+            # the overridden has highest priority
+            if config.triton.mix_order_reduction_split_size is not None:
+                return config.triton.mix_order_reduction_split_size
+
+            # heuristics based on number of SMs
+            device_prop = DeviceProperties.create(node1.get_device())
+            num_sm = device_prop.multi_processor_count
+            estimated_num_splits = num_sm * 8
+            split_size = max(next_power_of_2(numel // estimated_num_splits), 16)
+            split_size = min(split_size, 128)
```

Comment:
why is there an upper bound here? i guess the goal is to avoid wave quantization?

Reply:
So inside each split, we handle each row in serial. I was a bit conservative and thought a heuristic picking too large split size may hurt something. But on the other hand, the split-size is computed to make sure we have decent number of thread blocks (num_sm * 8) , so probably no upper bound is also fine.